Templates/templates/PHP/Model/EntityType.php.tt

```diff
@@ -286,7 +286,7 @@ foreach(var property in entity.Properties.Where(prop => prop.Type.GetTypeString(
     {
         $serializableProperties = $this->getProperties();
         foreach ($serializableProperties as $property => $val) {
-            if (is_a($val, "\DateTime")) {
+            if (get_class((object)$val) === "\DateTime")) {
```

Comment:
`is_a()` would be better instead of casting to object. `is_a` also checks if an object has `DateTime` as its parent